File: react-demo/src/demo/States.js
import { useState } from "react"

function States() {
  const [count, setCount] = useState(0)
  const style = {
    marginTop: "16px"
  }

  return (
    <button style={style} onClick={() => setCount(count + 1)}>
      Count: {count}
    </button>
  )
}

export default States

File: ComposeDemo/app/src/main/java/com/example/composedemo/demo/States.kt
package com.example.composedemo.demo

import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

@Composable
fun States() {
    var count by remember { mutableIntStateOf(0) }

    Button(
        onClick = { count++ },
        modifier = Modifier.padding(16.dp)
    ) {
        Text("Count: $count")
    }
}